ndonnx/_array.py

```diff
@@ -371,8 +368,8 @@ def __bool__(self) -> bool:
             raise ValueError(f"Cannot convert Array of shape {self.shape} to a bool")
 
     def __len__(self) -> int:
-        if isinstance(eager_value := self.shape, tuple):
-            return eager_value[0]
+        if isinstance(self.shape[0], int):
+            return self.shape[0]
         else:
             raise ValueError(f"Cannot convert Array of shape {self.shape} to a length")
```

Comment:
This error is not very coherent, can we change it?

Reply:
Good point, and neither are `__bool__` and `__int__` error messages.